server/neptune/terraform/command.go

```diff
@@ -56,47 +66,72 @@ func newArgument(arg string) (Argument, error) {
 	}, nil
 }
 
-type CommandArguments struct {
-	Command     Operation
-	CommandArgs []Argument
-	ExtraArgs   []Argument
+type SubCommand struct {
+	op    Operation
+	args  []Argument
+	flags []Flag
 }
 
-func NewCommandArguments(command Operation, commandArgs []Argument, extraArgs ...Argument) (*CommandArguments, error) {
+func NewSubCommand(op Operation) *SubCommand {
+	return &SubCommand{
+		op: op,
+	}
+}
 
-	return &CommandArguments{
-		Command:     command,
-		ExtraArgs:   extraArgs,
-		CommandArgs: commandArgs,
-	}, nil
+// WithArgs dedups incoming args using a "last one wins" approach
+// and sets them appropriately on the receiver
+func (c *SubCommand) WithArgs(args ...Argument) *SubCommand {
+	c.args = c.dedup(args)
+	return c
 }
 
-func (t CommandArguments) Build() []string {
-	finalArgs := []string{}
-	usedExtraArgsIndex := map[int]bool{}
-	for _, arg := range t.CommandArgs {
-
-		overrideIndex := -1
-		for i, overrideArg := range t.ExtraArgs {
-			if overrideArg.Key == arg.Key {
-				usedExtraArgsIndex[i] = true
-				overrideIndex = i
-			}
-		}
+func (c *SubCommand) WithFlags(flags ...Flag) *SubCommand {
+	c.flags = flags
```

Comment:
should flags be deduped too?

Reply:
probably, but we don't support extra flags, so it's probably fine to punt.